wagtail/contrib/forms/forms.py

```diff
@@ -100,35 +111,15 @@ def get_formatted_field_choices(self, field):
         Split the provided choices into a list, separated by new lines.
         If no new lines in the provided choices, split by commas.
         """
-
-        if "\n" in field.choices:
-            choices = (
-                (
-                    x.strip().rstrip(",").strip(),
-                    x.strip().rstrip(",").strip(),
-                )
-                for x in field.choices.split("\r\n")
-            )
-        else:
-            choices = ((x.strip(), x.strip()) for x in field.choices.split(","))
-
-        return choices
+        return [(x, x) for x in _split_by_newline_or_comma(field.choices)]
```

Comment:
Note that this changes the return type of this method. It now returns a `list` (as was documented in the docstring) rather than an iterator.